### PR TITLE
Improve login error messages for no synced pgp key, bad passphrase

### PR DESCRIPTION
**Pull request by patrickxb**
r? @malgorithms

### CODE REVIEW

File: go/engine/login_provision.go [hunk 1]
```diff
@@ -566,6 +566,10 @@ func (e *LoginProvision) syncedPGPKey(ctx *Context) (libkb.GenericKey, error) {
 	return unlocked, nil
 }
 
+func (e *LoginProvision) noSyncedKeyMsg() string {
+	return fmt.Sprintf("ERROR!\n\nSorry, you have a PGP private key that Keybase doesn't have a copy of. You need to prove you're you. Any of these will unlock this computer:\n\n1. reset your account and start fresh: https://keybase.io/%s and click \"reset my keys\" in the gear\n2. install GPG and put your PGP private key on this machine and try again\n3. go back and provision with another device or paper key\n\n", e.user.GetName())
```
Comment (maxtaco) [inline go/engine/login_provision.go]: Can we move this output to `client/`? I try to keep U/I-related stuff there as much as possible.  We can import/export the error type and then switch on it...

Comment (patrickxb) [inline go/engine/login_provision.go]: Will need a different error type then as libkb.NoKeyError is used for other missing keys as well and then the client output would be incorrect...

Comment (maxtaco) [inline go/engine/login_provision.go]: Ok I think that makes sense.

**Comment by patrickxb** [comment]
PTAL

**Comment by malgorithms** [comment]
the wording looks good to me. no idea code-wise!

**Comment by maxtaco** [comment]
Ok looks good thanks for fixing. Please also reserve this constant in keybase/keybase/lib/status_codes

**Comment by patrickxb** [comment]
I did.

**Pull request merged**